lib/cipherstash/client.rb

```diff
@@ -31,6 +31,26 @@ def self.client_options
       Profile.profile_options
     end
 
+
+    # Log into a CipherStash workspace.
+    #
+    # If `workspace` is given, a profile will be created for that workspace.
+    # This is useful for logging into a workspace for the first time.
+    #
+    # If a workspace is not provided, this command will load an existing profile and then log in using that profile.
+    # The profile name defaults to "default", but can be overridden with the `CS_PROFILE_NAME` env var or the `defaultProfile` opt in `{cs_config_path}/config.json`.
+    #
+    # @return [TrueClass]
+    #
+    # @raise [CipherStash::Client::Error::CreateProfileFailure] if a workspace is given and a new profile could not be created.
+    #
+    # @raise [CipherStash::Client::Error::LoadProfileFailure] if a workspace is not given and an existing profile could not be loaded.
+    #
+    def self.log_in(workspace: nil, profile_name: nil, logger: default_logger)
+      Profile.log_in(workspace: workspace, profile_name: profile_name, logger: logger)
+      true
+    end
```

Comment:
What is the purpose of the return value?

Unless there's prior art I'm not aware of, can we call the method `login` instead of `log_in`?

Reply:
> What is the purpose of the return value?

There's not a meaningful return value so I went with `true` to indicate that the call succeeded. I mostly wanted to avoid returning the profile since it's not really needed. I went with `true` since that's the same behaviour as the `create_collection` method in this file.

> Unless there's prior art I'm not aware of, can we call the method login instead of log_in?

My rationale here was that logging in is an action (verb) so I went with `log_in`. This is also consistent with our writing style guide (that doesn't necessarily apply to code I suppose):

> Use sign in rather than log in (verb) for calls-to-action where users enter their details to access a service.
>
>Do not use login as a noun (like ‘enter your login to access the console’). Say what the user actually needs to enter (like ‘enter your email and password to access the console’).

`login` would be consistent with the Rake task in ActiveStash and it's probably not worth updating the name of the task and docs in various places, so I can understand going with `login` instead as well.